File: core/domain/src/main/java/com/example/core/domain/product/Product.kt
package com.example.core.domain.product

import java.time.ZoneId
import java.time.ZonedDateTime

data class Product (
  val id: String?,
  val name: String,
  val description: String?,
  val price: Double,
  val rentalPrice: Double?,
  val color: Long,
  val size: Double,
  val imageUrl: String?,
  val stock: Int = 0,
  val created: ZonedDateTime = ZonedDateTime.now()
    .withZoneSameInstant(ZoneId.of("UTC"))
)

File: Backend/internal/store/models/product.go
package models

type Product struct {
	BaseModel
	Name        string   `json:"name"`
	Description *string  `json:"description,omitempty"`
	Price       float64  `json:"price"`
	RentalPrice *float64 `json:"rental_price,omitempty"`
	Color       uint32   `json:"color"`
	Size        float64  `json:"size"`
	ImageURL    *string  `json:"image_url,omitempty"`
	Stock       int      `json:"stock"`
}
